### PR TITLE
feat: Add 2FA flow

### PR DESCRIPTION
**Pull request by rajsam003**
This PR resolves [MOPLAT-292]

https://user-images.githubusercontent.com/17421923/183638277-a46ed5be-a799-4725-875d-7de16dbcb689.mov

### PR Checklist

- [x] I tested my changes on **iOS** and **Android**.
- [x] I added screenshots or videos to illustrate my changes.
- [ ] I added Tests and Stories for my changes.

### To the reviewers 👀

- [x] I would like **at least one** of the reviewers to run this PR on the simulator or device.

Need help with something? Have a look at our [docs], or get in touch with us.

[docs]: /docs/README.md

[MOPLAT-292]: https://artsyproduct.atlassian.net/browse/MOPLAT-292?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/screens/Auth/Login.tsx [hunk 1]
```diff
@@ -1,32 +1,40 @@
...
 import { useSafeAreaInsets } from "react-native-safe-area-context"
 import * as Yup from "yup"
 import { GlobalStore } from "app/store/GlobalStore"
-import { Box, Button, Flex, Input, Spacer, Text, useColor, useSpace } from "palette"
+import { Box, Button, Flex, Input, Spacer, Text, useColor } from "palette"
 
 export interface LoginSchema {
   email: string
   password: string
+  otp: string
```
Comment (patrinoua) [inline src/app/screens/Auth/Login.tsx:12]: What does otp stand for?
Can we please avoid shortening the words and use full words for better clarity? 🙂

File: src/app/screens/Auth/Login.tsx [hunk 2]
```diff
@@ -139,27 +160,53 @@ export const LoginScreenContent: React.FC = () => {
             value={values.password}
             error={errors.password}
           />
+          {showOtpInputField ? (
```
Comment (patrinoua) [inline src/app/screens/Auth/Login.tsx:163]: Can we combine the two `{showOtpInputField ? (` ?

**Review by patrinoua — APPROVED** [review]
Great! 🎉

File: src/app/screens/Auth/Login.tsx (hunk 2 above)
Comment (rajsam003) [reply src/app/screens/Auth/Login.tsx:163]: I dont get it? You mean to combine the spacer and Input?

File: src/app/screens/Auth/Login.tsx (hunk 1 above)
Comment (pvinis) [reply src/app/screens/Auth/Login.tsx:12]: feels like on of those things that you learn once by googling and that's enough?

**Review by pvinis — APPROVED** [review]

**Pull request merged**